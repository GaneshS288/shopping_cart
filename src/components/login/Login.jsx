function Login() {
    return(
        <div>
            <form action="">
                <label htmlFor="username">
                    Username :
                    <input type="text" id="username" name="username" required/>
                </label>

                <button type="submit">Log in as User</button>
            </form>

            <button type="button">Log in as Guest</button>
        </div>
    );
}

export default Login;