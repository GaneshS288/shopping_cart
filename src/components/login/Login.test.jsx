import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Login from "./Login.jsx";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import userEvent from "@testing-library/user-event";

function testSetup() {
  const routes = [
    {
      path: "/login",
      element: <Login></Login>,
    },
  ];

  const router = createMemoryRouter(routes, {
    initialEntries: ["/", "/login"],
    initialIndex: 1,
  });

  const user = userEvent.setup();

  return { router, user };
}

describe("Login", () => {
  it("matches snapshot", () => {
    const { router } = testSetup();
    const { container } = render(
      <RouterProvider router={router}></RouterProvider>
    );

    screen.debug()

    expect(container).toMatchSnapshot();
  });

  it("renders the login page", () => {
    const { router } = testSetup();
    render(<RouterProvider router={router}></RouterProvider>);

    const userNameInput = screen.getByLabelText("Username :");
    const [loginAsUserButton, loginAGuestButton] = screen.getAllByRole("button");

    expect(userNameInput).toBeInTheDocument();
    expect(loginAsUserButton.textContent).toMatch("Log in as User");
    expect(loginAGuestButton.textContent).toMatch("Log in as Guest");
  });
});
