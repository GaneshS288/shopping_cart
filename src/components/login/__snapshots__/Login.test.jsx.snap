// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > matches snapshot 1`] = `
<div>
  <div>
    <form
      action=""
    >
      <label
        for="username"
      >
        Username :
        <input
          id="username"
          name="username"
          required=""
          type="text"
        />
      </label>
      <button
        type="submit"
      >
        Log in as User
      </button>
    </form>
    <button
      type="button"
    >
      Log in as Guest
    </button>
  </div>
</div>
`;
